common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -168,6 +171,20 @@ protected void serviceInit(Configuration conf) throws Exception {
       TransportConf transportConf = new TransportConf("shuffle", new HadoopConfigProvider(conf));
       blockHandler = new ExternalShuffleBlockHandler(transportConf, registeredExecutorFile);
 
+      // register metrics on the block handler into the Node Manager's metrics system.
+      try {
+        YarnShuffleServiceMetrics serviceMetrics = new YarnShuffleServiceMetrics(
+            blockHandler.getAllMetrics());
+
+        MetricsSystemImpl metricsSystem = (MetricsSystemImpl) DefaultMetricsSystem.instance();
+        metricsSystem.register("shuffleService", "Metrics on the Spark Shuffle Service",
+            serviceMetrics);
+        logger.info("Registered metrics with Hadoop's DefaultMetricsSystem");
+      } catch (Exception e) {
+        logger.warn("Unable to register Spark Shuffle Service metrics with Node Manager; " +
+            "proceeding without metrics", e);
+      }
```

Comment:
there is not need for try / catch here, as we're not using reflection anymore; also there is no need to retype MetricsSystem ->

```
      // register metrics on the block handler into the Node Manager's metrics system.
      final YarnShuffleServiceMetrics serviceMetrics =
          new YarnShuffleServiceMetrics(blockHandler.getAllMetrics());
      final MetricsSystem metricsSystem = DefaultMetricsSystem.instance();
      metricsSystem.register(
          "shuffleService", "Metrics on the Spark Shuffle Service", serviceMetrics);
      logger.info("Registered metrics with Hadoop's DefaultMetricsSystem");
```